exonum/src/storage/options.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Abstract settings for databases.
+
+/// Options for database.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct DbOptions {
+    /// Number of open files that can be used by the DB.
+    ///
+    /// Defaults to `-1`, which means opened files are always kept open.
```

Comment:
I understand that this is an implementation detail of RocksDB, but I don't like such "magic constants". What do you think about using `Options<u32>` here (and converting `None` to the `-1` when needed)?

Reply:
Sounds reasonable. Don't know if we would be able to implement abstract API for any storage though.